endaq/calc/psd.py

```diff
@@ -335,3 +336,263 @@ def vc_curves(
+            typing.Callable[[np.ndarray, int], float],
+        ] = "mean",
+        freq_splits: np.array = None,
+        add_resultant: bool = True,
+        disable_warnings: bool = True,
+        **kwargs,
+) -> pd.DataFrame:
+    """
+    Compute a spectrogram for a time series data set using :py:func:`scipy.signal.spectrogram()`
+
+    :param df: the input dataframe with an index defining the time in seconds or datetime
+    :param num_slices: the number of slices to split the time series into, default is 100
+    :param scaling: the scaling of the output; `"density"` & `"spectrum"`
+        correspond to the same options in ``scipy.signal.welch``; `"parseval"`
+        will maintain the "energy" between the input & output, s.t.
+        ``welch(df, scaling="parseval").sum(axis="rows")`` is roughly equal to
+        ``df.abs().pow(2).sum(axis="rows")``;
+        `"unit"` will maintain the units and scale of the input data. `"rms"` will use `"parseval"` for the PSD
+        calculations and set `agg` to "sum", but then take the square root at the end
+    :param bin_width: the bin width or resolution in Hz for the PSD or FFT, defaults to 1
+    :param octave_bins: the number of frequency bins in each octave, defaults to `None`
+    :param fstart: the lowest frequency for an octave spaced output, defaults to 1
+    :param agg: the method for aggregating values into bins (only used if converting to octave or jagged); `'mean'` preserves
+        the PSD's area-under-the-curve, `'sum'` preserves the PSD's "energy"
+    :param freq_splits: the boundaries of the frequency bins to pass to :py:func:`~endaq.calc.psd.to_jagged()`
+    :param add_resultant: if `True` (default) the root sum of squares of each PSD column will
+        also be computed
+    :param disable_warnings: if `True` (default) it disables the warnings in helper functions
+    :param kwargs: Other parameters to pass directly to :py:func:`scipy.signal.spectrogram()`
+    :return: a dataframe containing all the spectrograms "melted" with columns defining the value, frequency, timeslice,
+        and original column from the input dataframe
+
+    .. seealso::
+
+        - `SciPy Spectrogram method <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.spectrogram.html>`_
+          Documentation for the function wrapped internally.
+    """
+
+    # Apply Scaling
+    if scaling == "parseval":
+        kwargs["scaling"] = "density"
+    elif scaling == "unit":
+        kwargs["scaling"] = "density"
+    elif scaling == "rms":
+        kwargs["scaling"] = "density"
+        agg = "sum"
+    elif scaling is not None:
+        kwargs["scaling"] = scaling
+
+    # Windowing, Default to Boxcar
+    if "window" not in kwargs:
+        kwargs["window"] = "boxcar"
+
+    # Determine Scipy.Spectrogram Stats
+    fs = 1 / utils.sample_spacing(df)
+    nperseg = int(fs / bin_width)
+    num_f = int(len(df) / fs / bin_width)
+    noverlap = nperseg - (nperseg * num_f / num_slices)
```

Comment:
use integer division here to ensure data types
```suggestion
    noverlap = nperseg - (nperseg*num_f//num_slices)
```